primitives/src/validator.rs

```diff
@@ -17,14 +19,51 @@ pub enum ValidatorError {
 
 pub type ValidatorFuture<T> = Pin<Box<dyn Future<Output = Result<T, ValidatorError>> + Send>>;
 
-pub trait Validator {
-    fn tick(&self, channel: Channel) -> ValidatorFuture<()>;
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+#[serde(transparent)]
+pub struct ValidatorId(#[serde(with = "SerHex::<StrictPfx>")] [u8; 20]);
+
+impl ValidatorId {
+    pub fn inner(&self) -> [u8; 20] {
+        self.0
+    }
+
+    pub fn to_hex_non_prefix_string(&self) -> String {
+        hex::encode(self.0)
+    }
+    pub fn to_hex_checksummed_string(&self) -> String {
+        eth_checksum::checksum(&format!("0x{}", hex::encode(self.0)))
+    }
+}
+
+impl TryFrom<&str> for ValidatorId {
+    type Error = DomainError;
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        let hex_value = if value.len() == 42 {
+            &value[2..]
+        } else {
+            value
+        };
+
+        let result = hex::decode(hex_value).map_err(|_| {
+            DomainError::InvalidArgument("Failed to deserialize validator id".to_string())
+        })?;
+        let mut id: [u8; 20] = [0; 20];
+        id.copy_from_slice(&result[..]);
+        Ok(Self(id))
+    }
+}
+
+impl fmt::Display for ValidatorId {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", format!("0x{}", hex::encode(self.0)))
```

Comment:
You can use the SerHex btw :wink: 
https://github.com/fspmarshall/serde-hex/issues/3#issuecomment-497826234